ipfs/crypto.go

```diff
@@ -34,6 +37,32 @@ func IdentityConfig(sk libp2pc.PrivKey) (config.Identity, error) {
 	return ident, nil
 }
 
+// UnmarshalPrivateKey converts a protobuf serialized private key into its
+// representative object
+func UnmarshalPrivateKey(data []byte) (libp2pc.PrivKey, error) {
```

Comment:
For future, may as well _always_ use this to unmarshal our ed25519 keys for backwards compat.

Reply:
Yeh good call. I can also port this over the JS land when required.